### DISCUSSION TITLE
vim `Copilot: Not authorised`

### DISCUSSION THREAD

**Discussion by HenryDayHall** [community/community#60600] [Copilot Conversations] [answered]
### Select Topic Area

Debugging.

### Body

So I think it's the same issue as https://github.com/orgs/community/discussions/41139 

I'm running MacOS; 13.2.1 (22D68) Ventura.

My copilot version is `1.9.1`, Node.js is version `20.4.0`.

My version of vim is 9.0.1677 but interestingly, `Copilot version` reports it as `Vim 9.0.0677`. Tried reinstalling both copilot and vim to see if they could be encouraged to agree, but they remained mismatched. Tried 2 versions of vim (one from homebrew, one that I compiled myself).

If I do `Copilot setup` it returns `Copilot: Authenticated as GitHub used HenryDayHall`, but then `Copilot status` gives `Copilot: Not authorised`.

1. Can anyone report a combination of versions for copilot.vim, node.js and vim that works for you?
2. Is there anymore debugging information I can get out of this?

Edit - resolved.
The problem was that the settings in https://github.com/settings/copilot were incomplete.

**Comment by OshekharO** [comment]
Try clearing the Copilot cache by removing the ~/.vim/copilot directory. This will force Copilot to re-authenticate and may resolve any authentication-related issues.

Also ensure that you are on latest version of Copilot.

**Reply by HenryDayHall** [reply, marked as answer]
The problem was that one of the settings in https://github.com/settings/copilot was incomplete.
Now resolved.

For what it's worth, mac OS doesn't seem to store the vim cache there.
